codestodians/food-basket-api
# [Ft:154589635] Add social authentication

#### What does this PR do?
- Add social Authentication mechanism

#### Description of Task to be completed
- Make it possible for users to log in with available social auth such as facebook, google

#### How should this be manually tested?

- run `pip3 install -r requirements.txt`
- run `python3 manage.py migrate`
run `gunicorn config.wsgi --reload`
- Set the newly added variables as listed in the .env.example file. 
  - Create a test app from [here](https://developers.facebook.com/tools/accesstoken) and generate a test user access token
-  run sample mutation with `mutation SocialAuth($provider: String!, $accessToken: String!) {
  socialAuth(provider: $provider, accessToken: $accessToken) {
    social {
      uid
    }
    token
  }
}`
see [this site](https://github.com/flavors/django-graphql-social-auth) for more context


#### What are the relevant pivotal tracker stories?
[#154589635](https://www.pivotaltracker.com/story/show/154589635)
